### PR TITLE
Add CI for .NET template

### LINKED ISSUE #19: CI for .NET

### PR DESCRIPTION
**Pull request by clbarrell**
This PR adds a CI for .NET template.

It uses docker for managing the .NET runtime and artifacts for passing around restored dependencies. 

Resolves #19

### CODE REVIEW

**Review by olyism — DISMISSED** [review]

**Comment by olyism** [comment]
hi @dannymidnight dotnet ready for review

File: dotnet-web/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+author: Buildkite
+---
+
+# CI for .NET with Buildkite
+
+This template gives you a continuous integration (CI) pipeline that automates installing dependencies, run unit test and build.
+
+At a glance:
+
+- For .NET projects
```
Comment (dannymidnight) [inline dotnet-web/README.md:14]: Following the others, let's link to the relevant .NET site.

File: dotnet-web/README.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+- Requires [Docker](https://docs.docker.com/get-docker/)
+- Automates build, test, and deployment
```
Comment (dannymidnight) [inline dotnet-web/README.md:16]: I reckon we need to be more specific on what this pipeline is using for building, testing, (and deploying). Is it deploying? It's not, we should remove that part.

File: dotnet-web/pipeline.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+steps:
+  - label: ":package: Restore dependencies"
```
Comment (dannymidnight) [inline dotnet-web/pipeline.yaml:2]: Maybe the `:dotnet:` emoji is the way to go? 

https://github.com/buildkite/emojis

File: dotnet-web/pipeline.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    command: "dotnet restore"
+    plugins:
+      - docker#v5.9.0:
+          image: "mono:latest"
```
Comment (dannymidnight) [inline dotnet-web/pipeline.yaml:6]: We should reference this `mono` docker image somewhere in the README. It wasn't immediately obvious to me what it was.

File: dotnet-web/pipeline.yaml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+  - wait
+
+  - label: ":construction_worker: Build"
+    command: "dotnet build --no-restore"
+    plugins:
+      - docker#v5.9.0:
+          image: "mono:latest"
```
Comment (dannymidnight) [inline dotnet-web/pipeline.yaml:22]: Seeing a missing new line here makes me think we should run prettier over the pipeline yaml. I'll make a note!

**Review by dannymidnight — CHANGES_REQUESTED** [review]

File: dotnet-web/pipeline.yaml (hunk 4 above)
Comment (olyism) [reply dotnet-web/pipeline.yaml]: Newbie mistake from my end 😅 there's no need for this so I've removed it.

File: dotnet-web/README.md (hunk 1 above)
Comment (olyism) [reply dotnet-web/README.md]: Updated README and added links https://github.com/buildkite/templates/pull/22/commits/bda5b8ca6cb30710b373092fc3d8d8d858b73673#diff-5fca869b4e5d00d2ce2ec8c059f2de58ea9e0e3cb3cd7f311bf0749d3c28098fR14

File: dotnet-web/README.md (hunk 2 above)
Comment (olyism) [reply dotnet-web/README.md]: Specified this pipeline uses the .NET SDK CLI to run restore, build and test.

**Comment by dannymidnight** [comment]
OK cool. I've taken over this PR and it's now dockerized with dependencies being passed from the first step to subsequent steps–allowing this pipeline to be run across multiple agents.

**Pull request merged**